tests/jballerina-unit-test/src/test/resources/test-src/record/readonly_record_fields_negative.bal

```diff
@@ -284,3 +284,20 @@ type RecordWithReadOnlyFields record {|
 
 RecordWithReadOnlyFields & readonly r1 = {x: []};
 readonly & RecordWithReadOnlyFields & readonly r2 = 1;
+
+type R1 record {
+    never x?;
+};
+
+type R2 record {|
+    never x?;
+    string y;
+|};
+
+function testRecordReadonlynessWithNeverFields() {
+    R1 r1 = {};
+    R2 r2 = {y: "hello"};
+
+    readonly x = r1;
+    readonly y = r2;
+}
```

Comment:
Shall we add one with an an anonymous record also?

Reply:
Added.